### PR TITLE
ATMOS Pipe Repair and Replacement

### PR DESCRIPTION
**Pull request by Azrun**
[feature][input wanted][balance]

## About the PR 

1. Adding new state for simple atmos pipes. "Destroyed" 
   * All contents of the pipe segment (gas) will be passed to the turf/tile.
   * "Destroyed" state becomes the more likely state when **exploded** instead of pipe removal.
   * Destroyed state becomes a possible state when pipe pressure exceeds **10 times** the fatigue pressure.
image

2. Destroyed state requires the piping segment be replaced by rods. 
   * 15 second duration (~7 seconds with carpenter or engineering training)
rebuild_piperebuild_pipe_2
   * 4 welds to fully repair
rebuild_pipe_3

3. Ruptured pipe **can** continue to rupture to a higher state. 

## Why's this needed? 
Addresses issues and complaints about engineering getting bombed by providing engineering the capability of replacing severely damaged sections of pipe.

## Changelog

```changelog
(u)Azrun
(*)Atmospheric pipes have a new severely damaged state that requires replacement by rods (and then welding).  Reduced chance of removing pipes from explosions in favor of new state.  Can also achieve new state through extreme pressure...
```

### CODE REVIEW

**Comment by Azrun** [comment]
🤦 Need to play with pipe code to actually disconnect nodes due to how that logic is abstracted from reality, then mingling case won't be necessary.

**Comment by pali6** [comment]
Straigt up buildable pipes would be better (but also a large(r) project), though this is idea is seems fine as a temporary fix

**Comment by Azrun** [comment]
> Straight up buildable pipes would be better (but also a large(r) project), though this is idea is seems fine as a temporary fix

Most of this change should be reverted/deleted it/when @ryebek or whoever actually gets around to constructable atmos. Exception being allowing rupture to escalate. 😈

**Comment by predmetparfuyit86** [comment]
> > Straight up buildable pipes would be better (but also a large(r) project), though this is idea is seems fine as a temporary fix
> 
> Most of this change should be reverted/deleted it/when @ryebek or whoever actually gets around to constructable atmos. Exception being allowing rupture to escalate. 😈

i think rybek has stopped working on that iirc

**Comment by Azrun** [comment]
Added more "surface" to the destroyed pipes.  Improves ability to interact and place rods.  

Removed some "surface" from replaced pipes. Improves indication that additional action is required.

File: code/modules/atmospherics/machinery/pipe.dm [hunk 1]
```diff
@@ -392,10 +443,14 @@ obj/machinery/atmospherics/pipe
...
 		pipeline_expansion()
+			if(destroyed)
+				return list(null, null)
 			return list(node1, node2)
 
 		update_icon()
-			if(node1&&node2)
+			if(destroyed)
+				icon_state = "destroyed"
+			else if(node1&&node2)
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/pipe.dm:453]:
```suggestion
			else if(node1 && node2)
```

File: code/modules/atmospherics/machinery/pipe.dm [hunk 2]
```diff
@@ -392,10 +443,14 @@ obj/machinery/atmospherics/pipe
 			..()
 
 		pipeline_expansion()
+			if(destroyed)
+				return list(null, null)
 			return list(node1, node2)
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/pipe.dm:448]:
```suggestion
			. = list(node1, node2)
			if(destroyed)
				return list(null, null)
```

File: code/modules/atmospherics/machinery/pipe.dm [hunk 3]
```diff
@@ -339,20 +344,35 @@ obj/machinery/atmospherics/pipe
 
 			return
 
-		proc/rupture(pressure)
-			if (pressure > 4*fatigue_pressure && prob(30)) ruptured = 3
-			else if (pressure > 2*fatigue_pressure && prob(60)) ruptured = 2
-			else ruptured = 1
+		proc/rupture(pressure, destroy=FALSE)
+			var/new_rupture
+			if (src.destroyed || destroy || (pressure > 10*fatigue_pressure && prob(1)) )
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/pipe.dm:349]: ambig. grouping
```suggestion
			if (src.destroyed || destroy || ((pressure > (10 * fatigue_pressure)) && prob(1)))
```

File: code/modules/atmospherics/machinery/pipe.dm [hunk 4]
```diff
@@ -339,20 +344,35 @@ obj/machinery/atmospherics/pipe
...
+				ruptured = 4
+				src.destroyed = TRUE
+				src.desc = "The remnants of a section of pipe that needs to be replaced.  Perhaps rods would be sufficient?"
+				parent.mingle_with_turf(loc, volume)
+				node1?.disconnect(src)
+				node2?.disconnect(src)
+				update_icon()
+				return
+			else if (pressure > 4*fatigue_pressure && prob(30)) new_rupture = 3
+			else if (pressure > 2*fatigue_pressure && prob(60)) new_rupture = 2
```
Comment (ZeWaka) [inline code/modules/atmospherics/machinery/pipe.dm:359]:
```suggestion
			else if ((pressure > (4 * fatigue_pressure)) && prob(30)) new_rupture = 3
			else if ((pressure > (2 * fatigue_pressure)) && prob(60)) new_rupture = 2
```

**Review by ZeWaka — APPROVED** [review]

File: code/modules/atmospherics/machinery/pipe.dm (hunk 1 above)
Comment (Azrun) [reply code/modules/atmospherics/machinery/pipe.dm]: Propagated suggestion.

**Comment by Azrun** [comment]
Testing functionality post addressing nits allowed me to find and address a bug! 🥳

**Pull request merged**